ni_measurementlink_service/session_management.py

```diff
@@ -138,27 +201,40 @@ def unreserve(self):
         """Unreserve sessions."""
         self._session_manager._unreserve_sessions(self._session_info)
 
+
+class SingleSessionReservation(BaseReservation):
+    """Manages reservation for a single session."""
+
+    @cached_property
+    def session_info(self) -> SessionInformation:
+        """Single session information object."""
+        assert len(self._session_info) == 1
+        return _convert_session_info_from_grpc(self._session_info[0])
+
+
+class MultiSessionReservation(BaseReservation):
+    """Manages reservation for multiple sessions."""
+
     @cached_property
     def session_info(self) -> List[SessionInformation]:
```

Comment:
I originally wanted to deprecate this property and rename it to `session_infos`, but that would affect 100% of users. @dixonjoel and I discussed this earlier and I'm going to leave this property alone for multi-session reservations.